### ISSUE TITLE
🐛 The Header does not accept a size property (It works, but the property is not available through TypeScript)

### ISSUE THREAD

**Issue by alexasselin008** [gsoft-inc/sg-orbit#609]
## Describe the bug

In this example, it shows how to specify a size to the accordion header.
https://orbit.sharegate.design/?path=/docs/accordion--default-story#size

However, the Header component does not accept a size in it's typescript interface. Should the component used here be AccordionHeader?

## Steps to reproduce

```
<Stack>
    <Accordion>
        <Item key="mars">
            <Header size="sm" as="h3"> {/* <---------- */}
                Mars
            </Header>
            <Content>
                Mars is the fourth planet from the Sun and the second-smallest planet.
            </Content>
        </Item>
        <Item key="jupiter">
            <Header size="sm" as="h3">
                Jupiter
            </Header>
            <Content>
                Jupiter is the fifth planet from the Sun and the largest in the Solar System.
            </Content>
        </Item>
        <Item key="venus">
            <Header size="sm" as="h3">
                Venus
            </Header>
            <Content>
                Venus is the second planet from the Sun. It is named after the Roman goddess of love
                and beauty.
            </Content>
        </Item>
    </Accordion>
    <Accordion>
        <Item key="mars">
            <Header as="h3">Mars</Header>
            <Content>
                Mars is the fourth planet from the Sun and the second-smallest planet.
            </Content>
        </Item>
        <Item key="jupiter">
            <Header as="h3">Jupiter</Header>
            <Content>
                Jupiter is the fifth planet from the Sun and the largest in the Solar System.
            </Content>
        </Item>
        <Item key="venus">
            <Header as="h3">Venus</Header>
            <Content>
                Venus is the second planet from the Sun. It is named after the Roman goddess of love
                and beauty.
            </Content>
        </Item>
    </Accordion>
</Stack>
```

## Expected results

```
<Stack>
    <Accordion>
        <Item key="mars">
            <AccordionHeader size="sm" as="h3"> {/* <---------- */}
                Mars
            </Header>
            <Content>
                Mars is the fourth planet from the Sun and the second-smallest planet.
            </Content>
        </Item>
        <Item key="jupiter">
            <AccordionHeader size="sm" as="h3">
                Jupiter
            </Header>
            <Content>
                Jupiter is the fifth planet from the Sun and the largest in the Solar System.
            </Content>
        </Item>
        <Item key="venus">
            <AccordionHeader size="sm" as="h3">
                Venus
            </Header>
            <Content>
                Venus is the second planet from the Sun. It is named after the Roman goddess of love
                and beauty.
            </Content>
        </Item>
    </Accordion>
    <Accordion>
        <Item key="mars">
            <AccordionHeader as="h3">Mars</Header>
            <Content>
                Mars is the fourth planet from the Sun and the second-smallest planet.
            </Content>
        </Item>
        <Item key="jupiter">
            <Header as="h3">Jupiter</Header>
            <Content>
                Jupiter is the fifth planet from the Sun and the largest in the Solar System.
            </Content>
        </Item>
        <Item key="venus">
            <AccordionHeader as="h3">Venus</Header>
            <Content>
                Venus is the second planet from the Sun. It is named after the Roman goddess of love
                and beauty.
            </Content>
        </Item>
    </Accordion>
</Stack>
```

**Comment by patricklafrance** [comment]
No, AccordionHeader is the underyling component which shouldn't be used directly. It's intended to be used in use case were someone want to build a custom header component for the accordion.

Since Header is meant to be a generic placeholder, I am not sure what is the solution here. It's something that have been developed prior to the decisions to move to TS.

I'll think about it and come back to you.

**Comment by patricklafrance** [comment]
I am currently in contact with the design team to validate if an Accordion really need to support multiple sizes.

If so, I will move the size prop upward to the Accordion component. It not optimal anyway to repeat the size on every headers.

I also still consider adding a `useAccordion` hook to faciliate the creation of custom components having a behavior similar to an accordion but with a different look. I'll create a new issue if we go this way.

I'll keep you posted.

**Comment by alexasselin008** [comment]
I've changed my use case to use disclosure instead. The issue is still valid, but not a priority

**Comment by patricklafrance** [comment]
We don't need sizing anymore. We will remove the size property.

@fraincs could you please determine with the designer what the appropriate specs should be for a "single" sized Accordion?

**Comment by patricklafrance** [comment]
(it's not a priority btw, just dispatching the work, the branch could sit there for a few days)

**Comment by patricklafrance** [comment]
After further discussions with the designers, we will drop the Accordion style as we know it today and go for the following style:

![image](https://user-images.githubusercontent.com/794579/118172109-feeb6380-b3f9-11eb-98ef-7f12b157cd8c.png)

**Comment by patricklafrance** [comment]
@fraincs I removed the size prop from Accordion, adjusted the docs / chromatic tests accordingly. Also added the ability to have an optionnal counter in the Accordion header.

Could you take care of the styling after you're done with your current tasks?

Everything is available in the following branch: https://github.com/gsoft-inc/sg-orbit/tree/rework_accordion

**Comment by fraincs** [comment]
Fixed with #622
